llmfoundry/models/layers/ffn.py

```diff
@@ -62,27 +92,34 @@ def build_ffn(
     d_model: int,
     expansion_ratio: int,
     fc_type: str = 'torch',
+    ffn_hidden_size: Optional[int] = None,
     device: Optional[str] = None,
     bias: bool = True,
     **kwargs: Any,
 ) -> nn.Module:
     ffn_type = kwargs.pop('ffn_type')
-    if ffn_type == 'mptmlp':
+    if ffn_type in ['mptmlp', 'mptgeglu']:
         if len(kwargs) > 0:
             raise ValueError(
-                f'MPTMLP got an unexpected keyword argument: {kwargs}')
-        return MPTMLP(
+                f'MPTMLP (or MPTGeGLU) got an unexpected keyword argument: {kwargs}'
+            )
+        return FFN_CLASS_REGISTRY[ffn_type](
             d_model=d_model,
             expansion_ratio=expansion_ratio,
             fc_type=fc_type,
+            ffn_hidden_size=ffn_hidden_size,
             device=device,
             bias=bias,
         )
     elif ffn_type == 'te_ln_mlp':
         assert te is not None
+        if ffn_hidden_size is not None:
```

Comment:
this looks like the same check as in `MPTMLP`, except less complete because it doesn't have the int checking stuff. should this check just happen once in this function instead of duplicated?

Reply:
exp ratio only gets passed into `MPTMLP` and not into the te layer.
removing exp ratio from MPTMLP signature is a breaking change (and I'd want to have this check happen inside of MPTMLP)

Reply:
I'll create a `_resolve_ffn_hidden_and_exp_ratio` fn